decidim-core/lib/tasks/decidim_webpacker_tasks.rake

```diff
@@ -39,10 +39,86 @@ namespace :decidim do
         webpack-dev-server
       )
       system! "npm uninstall #{webpacker_packages.join(" ")}"
+
+      # Add the Browserslist configuration to the project
+      add_decidim_browserslist_configuration
+    end
+
+    desc "Upgrade Decidim dependencies in Rails instance application"
+    task upgrade: :environment do
+      raise "Decidim gem is not installed" if decidim_path.nil?
+
+      # Update JS dependencies
+      install_decidim_npm
+    end
+
+    def install_decidim_npm
+      decidim_npm_packages.each do |type, package|
+        if type == :dev
+          system! "npm i -D #{package}"
+        else
+          system! "npm i #{package}"
+        end
+      end
+    end
+
+    def add_decidim_browserslist_configuration
+      package = JSON.parse(File.read(rails_app_path.join("package.json")))
+      return if package["browserslist"].is_a?(Array) && package["browserslist"].include?("extends @decidim/browserslist-config")
+
+      package["browserslist"] = ["extends @decidim/browserslist-config"]
+      File.write(rails_app_path.join("package.json"), JSON.pretty_generate(package))
     end
 
     def decidim_path
-      @decidim_path ||= Pathname.new(Gem.loaded_specs["decidim"].full_gem_path) if Gem.loaded_specs.has_key?("decidim")
+      @decidim_path ||= Pathname.new(decidim_gemspec.full_gem_path) if Gem.loaded_specs.has_key?("decidim")
+    end
+
+    def decidim_gemspec
+      @decidim_gemspec ||= Gem.loaded_specs["decidim"]
+    end
+
+    def decidim_npm_packages
+      case decidim_gemspec.source
+      when Bundler::Source::Path
+        gem_path = decidim_gemspec.source.path
+        gem_path = Pathname.new(ENV["BUNDLE_GEMFILE"]).dirname.join(gem_path) if gem_path.relative?
+
+        # The packages folder needs to be copied to the application folder
+        # because the linked dependencies are not installed when packages
+        # are installed using file references outside the application root
+        # where the `package.json` is located at. For more information, see:
+        # https://github.com/npm/cli/issues/2339
+        FileUtils.cp_r("#{gem_path}/packages", rails_app_path)
+
+        {
+          dev: "./packages/dev",
+          prod: "./packages/all"
+        }
+      when Bundler::Source::Rubygems
+        if decidim_gemspec.version.to_s =~ /\.dev$/
+          # With the .dev version the package does not exist at NPM yet.
+          {
+            dev: "https://gitpkg.now.sh/mainio/decidim/packages_dev/dev?feature/split-npm-packages",
```

Comment:
💬 Should we remove the branch modifier and use `develop`?

Reply:
Yes, after this is merged (in another PR).